src/SubtitleOctopus.cpp

```diff
@@ -337,10 +346,12 @@ class SubtitleOctopus {
     }
 
 private:
-    buffer_t m_blend;
+    ReusableBuffer m_blend;
     RenderBlendResult m_blendResult;
 };
 
 int main(int argc, char *argv[]) { return 0; }
 
+#ifdef __EMSCRIPTEN__
 #include "./SubOctpInterface.cpp"
+#endif
```

Comment:
First of all, unrelated to the buffer refactor.
Second: Are there some tests or some other use for compiling this without emscripten? If not, I see no use in adding the guard in the first place.

Reply:
It allows you to compile the file alone, iirc.

Reply:
I know, but is there even any *useful* standalone program or tests you can build with this file? For some reason SubtitleOctopus.cpp has a main function, but all it does is to immediately exit, which is not remotely useful.  
If there is no use for compiling it without emscripten, then the guard is not necessary and even misleading as it suggest there is some use in compiling it without emscripten.
If, on the other hand, there are some hidden tests which use this standalone we should integrate them into our CI.

Reply:
I just use `g++ SubtitleOctopus.cpp` to see if it compiles without having to run the full build process.

This has the same purpose: https://github.com/libass/JavascriptSubtitlesOctopus/blob/de8888fe7f3fe5e52e50ece1407b96b68a244b5f/src/SubtitleOctopus.cpp#L13-L18

Reply:
`make -C src` would also do, but if it’s already partly guarded anyway .. oh well. At least mention in the commit-message that this is only to quickly check for compiler warnings/errors and there’s no use to the compiled program or otheriwse.